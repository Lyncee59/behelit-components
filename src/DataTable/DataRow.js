import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { theme } from '../Tools/interpolation'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background: ${theme('dataRowBackgroundColor')};
  cursor: ${(props) => props.disableHighlight ? 'default' : 'pointer'};

  &:hover { 
    background: ${(props) => props.disableHighlight ? theme('dataRowBackgroundColor') : theme('dataRowHighlightBackgroundColor')};
  }
`

const DataRow = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

DataRow.propTypes = {
  children: PropTypes.node.isRequired,
  disableHighlight: PropTypes.bool
}

DataRow.defaultProps = {
  disableHighlight: false
}

export default DataRow
