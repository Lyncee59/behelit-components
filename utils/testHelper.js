import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)