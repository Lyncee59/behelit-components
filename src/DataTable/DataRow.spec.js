import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import DataRow from './DataRow'

describe('DataRow', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<DataRow>MyDataRow</DataRow>)

    expect(tree.find(DataRow)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<DataRow disableHighlight>MyDataRow</DataRow>)

    expect(tree.find(DataRow)).toMatchSnapshot()
  })
})
