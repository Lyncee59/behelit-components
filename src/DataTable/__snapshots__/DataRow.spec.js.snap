// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRow should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background: #F3F3F3;
  cursor: pointer;
}

.c0:hover {
  background: #FFFFFF;
}

<DataRow
  disableHighlight={false}
>
  <styled.div
    disableHighlight={false}
  >
    <StyledComponent
      disableHighlight={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background: ",
              [Function],
              ";
  cursor: ",
              [Function],
              ";

  &:hover { 
    background: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        MyDataRow
      </div>
    </StyledComponent>
  </styled.div>
</DataRow>
`;

exports[`DataRow should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background: #F3F3F3;
  cursor: default;
}

.c0:hover {
  background: #F3F3F3;
}

<DataRow
  disableHighlight={true}
>
  <styled.div
    disableHighlight={true}
  >
    <StyledComponent
      disableHighlight={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background: ",
              [Function],
              ";
  cursor: ",
              [Function],
              ";

  &:hover { 
    background: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        MyDataRow
      </div>
    </StyledComponent>
  </styled.div>
</DataRow>
`;
